config/decoder.go

```diff
@@ -401,7 +401,7 @@ func (d *decoder) valueStruct(key string, target interface{}) error {
 		field := targetType.Field(i)
 
 		// Check for the private field
-		if field.PkgPath != "" && !field.Anonymous {
+		if field.PkgPath != "" || field.Anonymous {
```

Comment:
yes, it is a bug in config, that was uncovered by populating manager struct that has a private embedded serviceCore..
It is really hard to test it outside the config package :)